src/components/pages/traits/list/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { Box, Flex, Heading, Image, Select, SimpleGrid } from "@chakra-ui/react";
+import BlueLink from "@components/@core/blue-link";
+import BoxHeading from "@components/@core/layout/box-heading";
+import LocalLink, { useLocalRouter } from "@components/@core/local-link";
+import { getTraitIcon } from "@utils/media";
+import React, { useState } from "react";
+
+export default function TraitsListComponent({ data, filterKey }) {
+  const router = useLocalRouter();
+  const [filter, setFilter] = useState<string>(filterKey);
+
+  // ✅ Define groupedData type to prevent "unknown" errors
+  const groupedData: Record<string, { categoryName: string; traitsValuePairList: any[] }[]> = {};
+
+  const applyFilter = (category) => {
+    setFilter(category);
+    router.push("/traits/list", true, { filter: category }, true);
+  };
+
+  // ✅ Build groupedData correctly
+  data.forEach((item) => {
+    const [mainCategory, subCategory] = item.categoryName.split(" > ");
+
+    if (!groupedData[mainCategory]) {
+      groupedData[mainCategory] = [];
+    }
+
+    groupedData[mainCategory].push({
+      categoryName: subCategory,
+      traitsValuePairList:
+        filterKey === "All"
+          ? item.traitsValuePairList
+          : item.traitsValuePairList.filter(
+              (i) => i.traits.showInObservation === (filterKey === "Species" ? false : true)
+            )
+    });
+  });
+
+  return (
+    <div className="container mt">
+      <Box mb={4} width="100%">
+        <Select maxW="10rem" ml="auto" value={filter} onChange={(e) => applyFilter(e.target.value)}>
```

Comment:
I think we already have a generic select component in the codebase. Look it up once and try to reuse it if possible